protocol/xmrmaker/checks.go

```diff
@@ -171,3 +172,29 @@ func (s *swapState) checkContract(txHash ethcommon.Hash) error {
 
 	return nil
 }
+
+func (s *swapState) checkAndSetTimeouts(t0, t1 *big.Int) error {
+	s.setTimeouts(t0, t1)
+
+	if s.Backend.Env() == common.Development {
+		return nil
+	}
```

Comment:
Is there a way we can set this up to have the full body executed in all environments?

Reply:
yeah, it's just that the `allowableTimeDiff` is longer than the actual timeout for development, which is janky

Reply:
Just throwing out possible ideas:
* Divide by 20.  Your using 1/20th in stagenet/mainnet where the duration is an hour. For dev's 2 minutes, that would be 6 seconds.
* Have `SwapTimeoutFromEnvironment` return both the expected `T1-T0` duration as the max `T0` delta from the current time.

Reply:
I updated to just divide by 20, I worry that may be too short of a diff time, but I'll see from the CI